src/components/progress-report/workflow/progress-report-workflow.repository.ts

```diff
@@ -146,4 +152,79 @@ export class ProgressReportWorkflowRepository extends DtoRepository(
       changes: { status },
     });
   }
+
+  async getProjectMemberInfoByReportId(reportId: ID) {
+    const query = this.db
+      .query()
+      .match([
+        node('report', 'ProgressReport', { id: reportId }),
+        relation('in', '', ACTIVE),
+        node('engagement', 'Engagement'),
+        relation('in', '', 'engagement', ACTIVE),
+        node('project', 'Project'),
+        relation('out', '', 'member', ACTIVE),
+        node('member', 'ProjectMember'),
+        relation('out', '', 'user', ACTIVE),
+        node('user', 'User'),
+      ])
+      .optionalMatch([
+        node('user'),
+        relation('out', '', 'email', ACTIVE),
+        node('email', 'EmailAddress'),
+      ])
+      .match([
+        node('member'),
+        relation('out', '', 'roles', ACTIVE),
+        node('role', 'Property'),
+      ])
+      .return<{
+        id: ID;
+        email?: string;
+        role?: Role[];
+      }>(['user.id as id', 'email.value as email', 'role.value as role']);
+    return await query.run();
+  }
+
+  async getUserInfoByEmail(email: string) {
+    const query = this.db
+      .query()
+      .match([
+        node('email', 'EmailAddress', { value: email }),
+        relation('in', '', 'email', ACTIVE),
+        node('user', 'User'),
+      ])
+      .return<{
+        id?: ID;
+        email?: string;
+      }>(['user.id as id', 'email.value as email']);
+    return await query.run();
+  }
+
+  async getProjectInfoByReportId(reportId: ID) {
+    const query = this.db
+      .query()
+      .match([
+        node('report', 'ProgressReport', { id: reportId }),
+        relation('in', '', ACTIVE),
+        node('engagement', 'Engagement'),
+        relation('in', '', 'engagement', ACTIVE),
+        node('project', 'Project'),
+      ])
+      .match([
+        node('engagement'),
+        relation('out', '', ACTIVE),
+        node('language', 'Language'),
+      ])
+      .return<{
+        projectId: ID;
+        languageId: ID;
+      }>(['project.id as projectId', 'language.id as languageId']);
+    const result = await query.first();
+    if (!result) {
+      throw new NotFoundException(
+        'Could not find project or language by report id',
+      );
+    }
```

Comment:
This will never be false. I'd probably just not null assert it. Otherwise I'd throw a ServerException, NotFound is a user error.
```suggestion
    const result = await query.first()!;
```